Framework/DataHandling/src/SetSample.cpp

```diff
@@ -711,8 +712,14 @@ void SetSample::setSampleShape(API::ExperimentInfo &experiment, const Kernel::Pr
   // Try known shapes or CSG first if supplied
   if (args) {
     const auto refFrame = experiment.getInstrument()->getReferenceFrame();
-    const auto xml = tryCreateXMLFromArgsOnly(*args, *refFrame);
+    auto xml = tryCreateXMLFromArgsOnly(*args, *refFrame);
     if (!xml.empty()) {
+      std::vector<double> rotationMatrix = experiment.run().getGoniometer().getR();
+      if (rotationMatrix != std::vector<double>{1, 0, 0, 0, 1, 0, 0, 0, 1} && !sampleEnv) {
```

Comment:
The Matrix class provides a method to return an identity matrix `Matrix<T>::identityMatrix()`. Please can you create a matrix using that to compare against? You could just do `rotationMatrix.identityMatrix()` I think

Reply:
The rotationMatrix isn't of type Matrix, it's a std::vector of the 9 inputs so I'm not sure I can call .identityMatrix or compare it to rotationMatrix. Do you think it's worth changing all rotationMatrix's to type Matrix?